raiden/log_config.py

```diff
@@ -119,36 +141,94 @@ def add_greenlet_name(
     return event_dict
 
 
-def redactor(blacklist: Dict[Pattern, str]) -> Callable[[str], str]:
-    """Returns a function which transforms a str, replacing all matches for its replacement"""
+def redact_secret(original_line: str) -> str:
+    redacted = original_line
+    for redact in LOG_REDACT_PATTERNS:
+        redacted = redact.pattern.sub(redact.replace, redacted)
+    return redacted
+
+
+def format_traceback(exc_traceback: TracebackType) -> Iterator[str]:
+    for frame, lineno in walk_tb(exc_traceback):
+        filename = frame.f_code.co_filename
+        name = frame.f_code.co_name
+        line = linecache.getline(filename, lineno).strip()
 
-    def processor_wrapper(msg: str) -> str:
-        for regex, repl in blacklist.items():
-            if repl is None:
-                repl = "<redacted>"
-            msg = regex.sub(repl, msg)
-        return msg
+        yield f'  File "{filename}", line {lineno}, in {name}\n' f"{line}\n"
 
-    return processor_wrapper
 
+def format_exception_chain(
+    exc_value: BaseException, exc_traceback: Optional[TracebackType], seen: Set[int]
+) -> Iterator[str]:
+    """Recursively format the chain of exceptions for logging.
 
-def _wrap_tracebackexception_format(redact: Callable[[str], str]):
-    """Monkey-patch TracebackException.format to redact printed lines.
+    Notes:
 
-    Only the last call will be effective. Consecutive calls will overwrite the
-    previous monkey patches.
+    - This does not include the stack locals for compactness;
+    - It uses the standard library `linecache` module, but it does not force a
+      cache reset, which saves a few system calls and won't may not show teh
+      wrong stack traces if the files are being edited;
+    - It does not handle exceptions traces larger than the stack depth;
+    - It does not handle SyntaxError specially.
```

Comment:
What is this bullet point for? Why would SyntaxError need special handling?

Reply:
Every bullet point is for one difference in respect to the previous implementation, SyntaxError just happens to be specially handled with the previous version.